src/transformers/trainer.py

```diff
@@ -272,6 +272,23 @@ def _get_fsdp_ckpt_kwargs():
         return {}
 
 
+def safe_globals():
+    # Starting from version 2.4 PyTorch introduces a stricter check for the objects which
+    # can be loaded with torch.load(). Starting from version 2.6 loading with weights_only=True
+    # requires allowlisting of such objects.
+    # See: https://github.com/pytorch/pytorch/pull/137602
+    # See: https://pytorch.org/docs/stable/notes/serialization.html#torch.serialization.add_safe_globals
+    if version.parse(torch.__version__) < version.parse("2.4.0"):
+        return contextlib.nullcontext()
+
+    allowlist = [np.core.multiarray._reconstruct, np.ndarray, np.dtype]
+    # numpy >1.25 defines numpy.dtypes.UInt32DType, but below works for
+    # all versions of numpy
+    allowlist += [type(np.dtype(np.uint32))]
```

Comment:
Should I add any other numpy dtypes in the list? As of now I spotted only `np.unit32` in the Transformers list as the one needed.

Reply:
The only one I don't see from accelerate is `encode`, however if things pass here without it it's accelerate specific and we don't need to worry about it

Reply:
Transformer tests did pass on my side without adding `encode`. This indeed seems accelerate specific.